src/test/kotlin/MicronautAuthenticationTest.kt

```diff
@@ -0,0 +1,246 @@
+        val controller = setupTest("global_authentication_prohibited.yml")
+        val functionAnnotations = controller.flatMap { it.members }
+            .flatMap { (it as TypeSpec).funSpecs.flatMap { it.annotations } }
+
+        val functionParameter = controller.flatMap { it.members }
+            .flatMap { (it as TypeSpec).funSpecs.flatMap { it.parameters } }
+
+        assertThat(functionAnnotations).anySatisfy{member ->
+            assertThat(member.className.canonicalName).isEqualTo("io.micronaut.security.annotation.Secured")
+            assertThat(member.members).anyMatch{ it.toString() == "SecurityRule.IS_ANONYMOUS"}
+        }
+
+        assertThat(functionParameter).noneSatisfy{ member ->
+            assertThat(member.name).isEqualTo("authentication")
+        }
+    }
+
+    @Test
+    fun `ensure that global authentication set to any AND empty authentication will add the optional parameter`() {
+        val controller = setupTest("global_authentication_optional.yml")
+        val functionAnnotations = controller.flatMap { it.members }
+            .flatMap { (it as TypeSpec).funSpecs.flatMap { it.annotations } }
+
+        val functionParameter = controller.flatMap { it.members }
+            .flatMap { (it as TypeSpec).funSpecs.flatMap { it.parameters } }
+
+        assertThat(functionAnnotations).anySatisfy{member ->
+            assertThat(member.className.canonicalName).isEqualTo("io.micronaut.security.annotation.Secured")
+            assertThat(member.members).anyMatch{ it.toString() == "SecurityRule.IS_AUTHENTICATED, SecurityRule.IS_ANONYMOUS"}
+        }
+
+        assertThat(functionParameter).anySatisfy{ member ->
+            assertThat(member.name).isEqualTo("authentication")
+            assertThat(member.type.isNullable == true)
+        }
+    }
+
+    @Test
+    fun `ensure that no authentication defined will add no parameter`() {
+        val controller = setupTest("global_authentication_none.yml")
+        val functionAnnotations = controller.flatMap { it.members }
+            .flatMap { (it as TypeSpec).funSpecs.flatMap { it.annotations } }
+
+        val functionParameter = controller.flatMap { it.members }
+            .flatMap { (it as TypeSpec).funSpecs.flatMap { it.parameters } }
+
+        assertThat(functionAnnotations).noneSatisfy{member ->
+            assertThat(member.className.canonicalName).isEqualTo("io.micronaut.security.annotation.Secured")
+            assertThat(member.members).anyMatch{ it.toString() == "SecurityRule.IS_AUTHENTICATED"}
+            assertThat(member.members).anyMatch{ it.toString() == "SecurityRule.IS_ANONYMOUS"}
+        }
+
+        assertThat(functionParameter).noneSatisfy{ member ->
+            assertThat(member.name).isEqualTo("authentication")
+        }
+    }
+
+    @Test
+    fun `ensure that global security and operational security defined will set the correct parameters `() {
```

Comment:
**suggestion:** `ensure that both global security and operation-level security being defined will result in the correctly generated code`